plasmapy/formulary/lengths.py

```diff
@@ -270,7 +289,22 @@ def gyroradius(
 
     omega_ci = frequencies.gyrofrequency(B, particle)
 
-    return np.abs(Vperp) / omega_ci
+    if np.isscalar(lorentzfactor):
+        # if it is not finite aka not defined then calc from vperp
+        if not isfinite_lorentzfactor:
+            lorentzfactor = np.copy(lorentzfactor)
```

Comment:
```suggestion
```
You don't need this one since the next line overwrite the reference; if you were doing `lorentzfactor[:] = RelativisticBody...`, then sure.